src/core/qgsexpressioncontext.cpp

```diff
@@ -82,12 +82,11 @@ QgsExpressionContextScope::~QgsExpressionContextScope()
 
 void QgsExpressionContextScope::setVariable( const QString &name, const QVariant &value, bool isStatic )
 {
-  if ( mVariables.contains( name ) )
+  auto it = mVariables.find( name );
+  if ( it != mVariables.end() )
   {
-    StaticVariable existing = mVariables.value( name );
-    existing.value = value;
-    existing.isStatic = isStatic;
-    addVariable( existing );
```

Comment:
isn't the addVariable line missing?

Reply:
@3nids it's not needed -- we alter the value directly. Before we were:

1. lookup to find if the value already existed
2. lookup to retrieve the current value
3. making a new variable based on the current value, then calling addVariable to update the hash (incurring another lookup) 

Now we just lookup once and alter directly if it exists

Reply:
thanks for the explanation.